File: 백준/Gold/2230. 수 고르기/수 고르기.py
import sys
input = sys.stdin.readline

if __name__ == "__main__":
    n, m = map(int, input().split())
    arr = [int(input()) for _ in range(n)]
    arr.sort()

    start, end = 0, 0
    cur = 2000000001
    while start <= end and end < n:
        if arr[end] - arr[start] < m:
            end += 1
        else:
            cur = min(cur, arr[end] - arr[start])
            start += 1
    print(cur)

File: 백준/Gold/2230. 수 고르기/수 고르기.java
import java.util.*;
import java.io.*;

public class Main {
	static int n, m;
	static int[] arr;
	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		m = Integer.parseInt(st.nextToken());
		arr = new int[n];
		for (int i = 0; i < n; i++){
			int tmp = Integer.parseInt(br.readLine());
			arr[i] = tmp;
		}
		Arrays.sort(arr);
		int l = 0;
		int r = 0;
		int result = Integer.MAX_VALUE;
		while (r < n){
			int tmp = arr[r] - arr[l];
			if (tmp < m){
				r++;
				continue;
			} else if (tmp == m) {
				result = tmp;
				break;
			}
			else{
				result = Math.min(result, tmp);
				l++;
			}
		}
		System.out.println(result);
	}

}